core/src/main/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTask.scala

```diff
@@ -79,12 +83,68 @@ class BackgroundTask(
 
     (Stream.sleep[IO](config.syncCloudStorageDirectoryInterval) ++ Stream.eval(res)).repeat
   }
+
+  val delocalizeBackgroundProcess: Stream[IO, Unit] = {
+    if (config.isRstudioRuntime) {
+      val res = for {
+        storageLinks <- storageLinksCache.get
+        implicit0(tid: Ask[IO, TraceId]) <- IO(TraceId(UUID.randomUUID().toString)).map(tid => Ask.const[IO, TraceId](tid))
+        _ <- storageLinks.values.toList.traverse { storageLink =>
+          if (storageLink.pattern.toString.contains(".Rmd")) {
+            findFilesWithPattern(config.workingDirectory.resolve(storageLink.localBaseDirectory.path.asPath), storageLink.pattern).traverse_ { file =>
+              val gsPath = getGsPath(storageLink, new File(file.getName))
+              val localAbsolutePath = config.workingDirectory.resolve(storageLink.localBaseDirectory.path.asPath).resolve(file.getName)
+              for {
+                sd <- shouldDelocalize(gsPath, localAbsolutePath)
+                _ <- if (sd) {
+                  googleStorageAlg.fileToGcs(
+                    RelativePath(java.nio.file.Paths.get(file.getName)),
+                    gsPath
+                  )
+                } else {
+                  IO.unit
+                }
+              } yield ()
+
+            }
+          } else IO.unit
+        }
+      } yield ()
+      (Stream.sleep[IO](2 seconds) ++ Stream.eval(res)).repeat
+    } else {
+      Stream.eval(logger.info("Not running rmd sync process because this is not an Rstudio runtime"))
+    }
+  }
+
+  def shouldDelocalize(gsPath: GsPath, localAbsolutePath: Path): IO[Boolean] =
```

Comment:
can you add a unit test for this...mocking out the `getObjectMetadata`'s response

Reply:
just pushed the 2 unit tests!